docs/docs/file-system-page-creation.md

```diff
@@ -0,0 +1,167 @@
+---
+title: Creating Pages from the File System
+---
+
+This page roughly documents the APIs and conventions available with the new file-based page creation features.
+
+For now, these features are marked as experimental and require a flag to utilize. Start your Gatsby site locally with the following flag:
+
+```bash
+GATSBY_EXPERIMENTAL_ROUTING_APIS=1 gatsby develop
+```
+
+## Creating client-only routes
+
+Client-only routes can be created by using square brackets (`[ ]`) in the file path to mark any dynamic segments of the URL. While Gatsby creates predictable routes for static pages, the same expectations apply to client side routes. Here are some examples:
+
+- `/src/pages/users/[id].js => /users/:id`
+- `/src/pages/products/[...splat].js => /users/*splat`
+- `/src/pages/app/[...].js => /app/*`
+
+The dynamic segment of the file name (the part between the square brackets) will be filled in and provided to your components on a `props.params` object. For example:
```

Comment:
Could you expand a little bit on the syntax here and lay out what `...` and `*` are for in this context? I'm guessing not everyone who wants to use this will be familiar.

Reply:
I have updated it, wanna check it out?